state/migration_export_test.go

```diff
@@ -687,6 +687,53 @@ func (s *MigrationExportSuite) TestOfferConnections(c *gc.C) {
 	c.Assert(offer.UserName(), gc.Equals, stOffer.UserName())
 }
 
+func (s *MigrationExportSuite) TestExternalControllers(c *gc.C) {
+	_, err := s.State.AddRemoteApplication(state.AddRemoteApplicationParams{
+		Name:        "gravy-rainbow",
+		URL:         "me/model.rainbow",
+		SourceModel: s.Model.ModelTag(),
+		Token:       "charisma",
+		OfferUUID:   "offer-uuid",
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	service := state.NewExternalControllers(s.State)
+	stCtrl, err := service.Save(crossmodel.ControllerInfo{
+		Addrs:         []string{"10.224.0.1:8080"},
+		Alias:         "magic",
+		CACert:        "magic-ca-cert",
+		ControllerTag: names.NewControllerTag("f47ac10b-58cc-4372-a567-0e02b2c3d479"),
+	}, s.Model.UUID(), "af5a9137-934c-4b0c-8317-643b69cf4971")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// We only care for the offer connections
+	model, err := s.State.ExportPartial(state.ExportConfig{
```

Comment:
We should make a helper for this that sets everything to true, and takes a `func (state.ExportConfig)` that sets the desired import action to false.

Reply:
Let me think about this...

Reply:
I follow this up in a mechanical PR.